src/Processors/Formats/OutputFormatWithUTF8ValidationAdaptor.h

```diff
@@ -24,37 +26,39 @@ class OutputFormatWithUTF8ValidationAdaptorBase : public Base
         }
 
         if (validate_utf8 && values_can_contain_invalid_utf8)
-        {
-            validating_ostr = std::make_unique<WriteBufferValidUTF8>(this->out);
-            ostr = validating_ostr.get();
-        }
-        else
-            ostr = &this->out;
+            validating_ostr = std::make_unique<WriteBufferValidUTF8>(*Base::getWriteBufferPtr());
     }
 
     void flush() override
     {
-        ostr->next();
-
         if (validating_ostr)
-            this->out.next();
+            validating_ostr->next();
+        Base::flush();
     }
 
     void finalizeBuffers() override
     {
         if (validating_ostr)
             validating_ostr->finalize();
+        Base::finalizeBuffers();
     }
 
     void resetFormatterImpl() override
     {
-        validating_ostr = std::make_unique<WriteBufferValidUTF8>(this->out);
-        ostr = validating_ostr.get();
+        LOG_DEBUG(&Poco::Logger::get("RowOutputFormatWithExceptionHandlerAdaptor"), "resetFormatterImpl");
```

Comment:
Looks like some kind of debugging had been left.

Reply:
Yeah, let's delete it

Reply:
https://github.com/ClickHouse/ClickHouse/pull/57899